server/src/postgis/nearest.rs

```diff
@@ -0,0 +1,347 @@
+//! This module contains functions for routing between nodes.
+use crate::grpc::server::grpc_server::{DistanceTo, NearestNeighborRequest, NodeType};
+
+use uuid::Uuid;
+
+/// Possible errors with path requests
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum NNError {
+    /// No path was found
+    NoPath,
+
+    /// Invalid start node
+    InvalidStartNode,
+
+    /// Invalid end node
+    InvalidEndNode,
+
+    /// Invalid limit
+    InvalidLimit,
+
+    /// Invalid range
+    InvalidRange,
+
+    /// Unsupported path type
+    Unsupported,
+
+    /// Could not get client
+    Client,
+
+    /// Unknown error
+    Unknown,
+}
+
+impl std::fmt::Display for NNError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            NNError::NoPath => write!(f, "No path was found."),
+            NNError::InvalidStartNode => write!(f, "Invalid start node."),
+            NNError::InvalidEndNode => write!(f, "Invalid end node."),
+            NNError::InvalidLimit => write!(f, "Invalid limit."),
+            NNError::InvalidRange => write!(f, "Invalid range."),
+            NNError::Unsupported => write!(f, "Unsupported path type."),
+            NNError::Client => write!(f, "Could not get client."),
+            NNError::Unknown => write!(f, "Unknown error."),
+        }
+    }
+}
+
+impl NearestNeighborRequest {
+    fn validate(&self) -> Result<(), NNError> {
+        if self.limit < 1 {
+            postgis_error!(
+                "(validate NearestNeighborRequest) invalid limit: {}",
+                self.limit
+            );
+            return Err(NNError::InvalidLimit);
+        }
+
+        if self.max_range_meters < 0.0 {
+            postgis_error!(
+                "(validate NearestNeighborRequest) invalid max range meters: {}",
+                self.max_range_meters
+            );
+            return Err(NNError::InvalidRange);
+        }
+
+        Ok(())
+    }
+}
+
+/// Get the nearest neighboring vertiports to a vertiport
+async fn nearest_neighbor_vertiport_source(
+    stmt: tokio_postgres::Statement,
+    client: deadpool_postgres::Client,
+    request: NearestNeighborRequest,
+) -> Result<Vec<tokio_postgres::Row>, NNError> {
+    let Ok(start_node_id) = Uuid::parse_str(&request.start_node_id) else {
+        postgis_error!("(nearest_neighbor_vertiport_source) could not parse start node id into UUID: {}", request.start_node_id);
+        return Err(NNError::InvalidStartNode);
+    };
+
+    match client
+        .query(
+            &stmt,
+            &[
+                &start_node_id,
+                &request.limit,
+                &(request.max_range_meters as f64),
+            ],
+        )
+        .await
+    {
+        Ok(results) => Ok(results),
+        Err(e) => {
+            println!(
```

Comment:
Should be: `postgis_error!(`

Reply:
https://github.com/Arrow-air/svc-gis/commit/2b93948539015c3003bde9fba7385a81e9e6f4dd